app/models/user.rb

```diff
@@ -442,28 +442,38 @@ def self.diaspora_id_host
   end
 
   def seed_aspects
-    self.aspects.create(:name => I18n.t('aspects.seed.family'))
-    self.aspects.create(:name => I18n.t('aspects.seed.friends'))
-    self.aspects.create(:name => I18n.t('aspects.seed.work'))
-    aq = self.aspects.create(:name => I18n.t('aspects.seed.acquaintances'))
-
-    if AppConfig.settings.autofollow_on_join?
-      begin
-        default_account = Person.find_or_fetch_by_identifier(AppConfig.settings.autofollow_on_join_user)
-        share_with(default_account, aq)
-      rescue DiasporaFederation::Discovery::DiscoveryError
-        logger.warn "Error auto-sharing with #{AppConfig.settings.autofollow_on_join_user}
+    aspects.create(name: I18n.t("aspects.seed.family"))
+    aspects.create(name: I18n.t("aspects.seed.friends"))
+    aspects.create(name: I18n.t("aspects.seed.work"))
+    acquaintances = aspects.create(name: I18n.t("aspects.seed.acquaintances"))
+
+    acquaintances.tap do |aq|
+      if AppConfig.settings.autofollow_on_join?
+        autofollow_user = AppConfig.settings.autofollow_on_join_user
+        autofollow_accounts = AppConfig.settings.autofollow_on_join_accounts
+        begin
+          follow_account(autofollow_user, aq) if autofollow_user.present?
+          autofollow_accounts.each {|user_id| follow_account(user_id, aq) } if autofollow_accounts.present?
+        rescue DiasporaFederation::Discovery::DiscoveryError
+          logger.warn "Error auto-sharing with #{AppConfig.settings.autofollow_on_join_user}
                      fix autofollow_on_join_user in configuration."
+        end
```

Comment:
So in here I'd see the logic should be:

1. Look for `autofollow_on_join_accounts`, ensure it's a list (`Array.wrap`), fallback to an empty array if it's absent.
2. Look for `autofollow_on_join_user`, if it's present add it to the `autofollow_on_join_accounts` array.
3. Remove duplicates from `autofollow_on_join_accounts` (`Array#uniq`)

Plus test cases for all the combinations, that is absence of both or either setting, duplicate accounts between the two settings, distinct accounts between the two settings.

Reply:
Resolved here [38e13f6](https://github.com/diaspora/diaspora/pull/8430/commits/38e13f6f764d86f037046702d7d3e68da4b7710a).